### PR TITLE
Auto Flip

### PR DESCRIPTION
**Pull request by mablack01**
## Summary

Implements the auto-recommend flips MVP feature, which automatically cycles through flip recommendations one-by-one into Flip Assist. When a user places a buy order, the system advances to the next recommendation. When a buy order completes, the sell side for that item is automatically queued.

This feature streamlines the flipping workflow by eliminating manual clicking through recommendations, allowing users to focus on executing trades rather than navigating the UI.

### New Features
- **Auto-Recommend Toggle Button** - Added "Auto" toggle button to Flip Finder panel header
- **Auto-Recommend Service** - Core queue management service (`AutoRecommendService.java`) that:
  - Maintains a queue of recommendations from Flip Finder
  - Automatically focuses the first recommendation into Flip Assist
  - Advances to next recommendation when user places a buy order
  - Queues sell side when buy order completes
  - Integrates with existing GE offers (filters out items already in GE)
  - Shows status updates in the panel
- **Config Toggle** - Added `enableAutoRecommend` config option under Flip Assistant section
- **Status Display** - Auto-recommend status label shows current progress and queue state

### Implementation Details

**New Files:**
- `AutoRecommendService.java` (381 lines) - Core service managing the auto-recommend queue and workflow

**Modified Files:**
- `FlipSmartConfig.java` - Added `enableAutoRecommend` config option
- `FlipSmartPlugin.java` - Wired up service lifecycle and GE event notifications:
  - Initialize/shutdown service
  - Notify service on buy order placed
  - Notify service on buy order completed
  - Notify service on sell order placed
  - Stop service on logout
- `FlipFinderPanel.java` - Added UI components:
  - Auto toggle button in header
  - Status label for queue updates
  - Toggle handler that starts/stops service

### Refactoring
- **Modified `clearFlipAssistFocusIfMatches()`** - Skip auto-clearing focus when auto-recommend is active (service manages focus transitions)

### Automated Tests
- Build passes successfully with no errors
- No new test warnings introduced

### Manual Testing
- [x] Verify "Auto" button appears in Flip Finder header
- [x] Toggle auto-recommend on with recommendations loaded
- [x] Verify first recommendation focuses into Flip Assist
- [x] Place a buy order and verify focus advances to next recommendation
- [x] Verify buy completion queues the sell side
- [ ] Place sell order and verify service continues cycling
- [x] Verify items already in GE are filtered out
- [x] Toggle auto-recommend off mid-session
- [x] Verify service stops on logout
- [x] Verify status label updates throughout workflow

## API Changes

No API changes - this is RuneLite plugin code only.

## Frontend Impact

No frontend impact - this is RuneLite plugin functionality.

## Database Migrations

No database migrations required.

## Deployment Notes

- [ ] No environment variables required
- [ ] No dependencies added
- [ ] No configuration changes required
- [ ] No database migrations

## Acceptance Criteria Coverage

This PR implements the following acceptance criteria from issue Flip-Smart/flip-smart#223:

- **AC1** - Display recommendation in Flip Assist window
- **AC2** - Buy offer creation workflow (auto-focus buy step)
- **AC3** - Sell offer creation workflow (auto-focus sell step after buy completes)
- **AC5** - Basic integration with existing GE offers (filters out items already in GE)

**Not included in this PR** (future work):
- AC4 - Pausing auto-recommend (will be implemented in follow-up)
- AC6 - Advanced settings (queue size, filters)
- AC7 - Error handling and edge cases
- AC8 - Performance monitoring

## Checklist

- [x] Tests pass (build successful)
- [x] Code follows RuneLite plugin-hub guidelines (no thread interrupts on managed threads)
- [x] Commits are clean and descriptive
- [x] No debug code left
- [x] Types are correct (Lombok used appropriately)

## Related Issues

Closes Flip-Smart/flip-smart#223

### Service Architecture
The `AutoRecommendService` uses a queue-based architecture:
1. When started, loads current recommendations into internal queue
2. Immediately focuses first recommendation via callback
3. Listens to GE events from plugin (buy placed, buy completed, sell placed)
4. Advances queue based on user actions
5. Manages "pending sells" queue for items that finished buying

### Thread Safety
All state mutations happen on the main thread via callbacks. No explicit threading beyond what RuneLite provides.

### Integration Points
- **FlipSmartPlugin** - Wires up GE event notifications
- **FlipFinderPanel** - Provides UI controls and status display
- **FlipAssistOverlay** - Receives focus updates via callback (same pattern as manual focus)

### Screenshots
Screenshot 2026-02-12 at 7 30 10 PM

### CODE REVIEW

File: src/main/java/com/flipsmart/FlipSmartConfig.java [hunk 1]
```diff
@@ -345,6 +345,18 @@ default int priceOffset()
...
 	}
 
+	@ConfigItem(
+		keyName = "enableAutoRecommend",
+		name = "Auto-Recommend",
+		description = "Automatically cycle through flip recommendations one by one into Flip Assist",
+		section = flipAssistantSection,
+		position = 5
+	)
+	default boolean enableAutoRecommend()
```
Comment (brettkrueger) [inline src/main/java/com/flipsmart/FlipSmartConfig.java]: The `enableAutoRecommend` config option is defined here but never actually checked anywhere. The Auto button shows up unconditionally regardless of this setting. Either wire it up to control the button visibility or remove it so it's not confusing.

**Review by brettkrueger — APPROVED** [review]
Nice work on the fixes!
